### PR TITLE
Cecilidw quickstarts

### PR DESCRIPTION
**Pull request by cecilidw**
# PR into Azure/azureml-examples

## Checklist

I have:

-  [x] read and followed the contributing guidelines
-  [x] ran `python readme.py` after making changes (per contributing guidelines)
-  [x] added required testing (per contributing guidelines)

## Changes

-

fixes #

### CODE REVIEW

File: tutorials/quickstarts/AzureMLin10mins.ipynb [hunk 1]
```diff
@@ -0,0 +1,642 @@
...
+        "It is often the case that once you have your machine learning code working in a development environment that you want to productionize this by using the Python SDK and run the code as a **_job_** - ideally on a schedule or trigger (for example, arrival of new data). To this end, we recommend that you follow the next quickstart in this series, [**Learn how to get started with Azure ML Job Submission**](GettingStartedWithPythonSDK.ipynb). This quickstart is focused on running jobs-based machine learning code in the cloud."
+      ],
+      "metadata": {}
+    }
+  ],
+  "metadata": {
+    "kernel_info": {
+      "name": "python3-azureml"
+    },
+    "kernelspec": {
```
Comment (lostmygithubaccount) [inline tutorials/quickstarts/AzureMLin10mins.ipynb:619]: `readme.py` sets the kernelspec, which then conflicts with python 3.6...

File: tutorials/quickstarts/ClassificationWithAutomatedML.ipynb [hunk 2]
```diff
@@ -0,0 +1,504 @@
+{
+  "cells": [
+    {
+      "cell_type": "markdown",
+      "source": [
+        "Copyright (c) Microsoft Corporation. All rights reserved. \r\n",
```
Comment (lostmygithubaccount) [inline tutorials/quickstarts/ClassificationWithAutomatedML.ipynb:6]: no license cell needed

File: tutorials/quickstarts/ClassificationWithAutomatedML.ipynb [hunk 3]
```diff
@@ -0,0 +1,504 @@
...
+          "transient": {
+            "deleting": false
+          }
+        }
+      }
+    },
+    {
+      "cell_type": "code",
+      "source": [
+        "data = \"https://automlsamplenotebookdata.blob.core.windows.net/automl-sample-notebook-data/creditcard.csv\"\n",
```
Comment (lostmygithubaccount) [inline tutorials/quickstarts/ClassificationWithAutomatedML.ipynb:136]: I don't personally like using random old blob storage locations - we should converge these and have clear ownership

File: tutorials/quickstarts/GettingStartedWithPythonSDK.ipynb [hunk 4]
```diff
@@ -0,0 +1,715 @@
+{
+  "cells": [
+    {
+      "cell_type": "markdown",
+      "source": [
+        "Copyright (c) Microsoft Corporation. All rights reserved. \r\n",
+        "\r\n",
```
Comment (lostmygithubaccount) [inline tutorials/quickstarts/GettingStartedWithPythonSDK.ipynb:7]: no license cell

File: tutorials/quickstarts/GettingStartedWithPythonSDK.ipynb [hunk 5]
```diff
@@ -0,0 +1,715 @@
...
+          }
+        }
+      }
+    },
+    {
+      "cell_type": "markdown",
+      "source": [
+        "An Environment defines Python packages, environment variables, and Docker settings that are used in machine learning experiments. Here you will be using a curated environment that has already been made available through the workspace. \n",
+        "\n",
+        "Read [this article](https://docs.microsoft.com/en-us/azure/machine-learning/how-to-use-environments) if you want to learn more about Environments and how to use them."
```
Comment (lostmygithubaccount) [inline tutorials/quickstarts/GettingStartedWithPythonSDK.ipynb:339]: 1. no "/en-us" in links
2. generally avoid having links like this (they break often)

File: tutorials/quickstarts/GettingStartedWithPythonSDK.ipynb [hunk 6]
```diff
@@ -0,0 +1,715 @@
...
+          "transient": {
+            "deleting": false
+          }
+        }
+      }
+    },
+    {
+      "cell_type": "markdown",
+      "source": [
+        "Create a [ScriptRunConfig](https://docs.microsoft.com/en-us/python/api/azureml-core/azureml.core.scriptrunconfig?preserve-view=true&view=azure-ml-py) object to specify the configuration details of your training job, including your training script, environment to use, and the compute target to run on. A script run configuration is used to configure the information necessary for submitting a training run as part of an experiment. \n",
```
Comment (lostmygithubaccount) [inline tutorials/quickstarts/GettingStartedWithPythonSDK.ipynb:379]: same comment as above

File: tutorials/quickstarts/GettingStartedWithPythonSDK.ipynb [hunk 7]
```diff
@@ -0,0 +1,715 @@
...
+          "transient": {
+            "deleting": false
+          }
+        }
+      }
+    },
+    {
+      "cell_type": "markdown",
+      "source": [
+        "if you want to cancel a run, you can follow [these instructions](https://aka.ms/aml-docs-cancel-run)."
```
Comment (lostmygithubaccount) [inline tutorials/quickstarts/GettingStartedWithPythonSDK.ipynb:505]: same comment as above

File: tutorials/quickstarts/README.md [hunk 8]
```diff
@@ -0,0 +1,18 @@
+# Quickstarts for Compute Instance
+
+description: Quickstarts for use in Compute Instance
+
+The notebooks in this tutorial will be used as getting started examples on the Notebooks landing page in AML Studio, and will be run in a Compute Instance.
+
+The tutorial consists of three notebooks:
+
+- [AzureMLin10mins.ipynb](AzureMLin10mins.ipynb)
+- [GettingStartedWithPythonSDK.ipynb](GettingStartedWithPythonSDK.ipynb)
```
Comment (lostmygithubaccount) [inline tutorials/quickstarts/README.md:10]: consider ordering alphabetically (this is how they'll be ordered in the main README)

File: tutorials/quickstarts/README.md [hunk 9]
```diff
@@ -0,0 +1,18 @@
...
+- [ClassificationWithAutomatedML.ipynb](ClassificationWithAutomatedML.ipynb)
+
+**AzureML in 10mins** 
+This notebook takes the user through a data ingest, training and deployment workflow that can be completed in 10minutes. This notebook does *not* use job submission, but rather an interactive run to reduce the number of concepts.
```
Comment (lostmygithubaccount) [inline tutorials/quickstarts/README.md:14]: add an empty line between the bold header and text

File: tutorials/quickstarts/automl_setup_linux.sh [hunk 10]
```diff
@@ -0,0 +1,66 @@
+#!/bin/bash
```
Comment (lostmygithubaccount) [inline tutorials/quickstarts/automl_setup_linux.sh:1]: this file is scary

File: tutorials/quickstarts/check_conda_version.py [hunk 11]
```diff
@@ -0,0 +1,28 @@
+from distutils.version import LooseVersion
```
Comment (lostmygithubaccount) [inline tutorials/quickstarts/check_conda_version.py:1]: is this file needed?

File: tutorials/quickstarts/environment.yml [hunk 12]
```diff
@@ -0,0 +1,29 @@
+name: azureml_py36
+dependencies:
+  # The python interpreter version.
+  # Currently Azure ML only supports 3.5.2 and later.
```
Comment (lostmygithubaccount) [inline tutorials/quickstarts/environment.yml:4]: is all the pinning needed? that usually indicates a pretty fragile example

File: tutorials/quickstarts/environment.yml [hunk 13]
```diff
@@ -0,0 +1,29 @@
...
+- holidays==0.9.11
+- pytorch::pytorch=1.4.0
+- cudatoolkit=10.1.243
+
+- pip:
+  # Required packages for AzureML execution, history, and data preparation.
+  - azureml-widgets~=1.22.0
+  - pytorch-transformers==1.0.0
+  - spacy==2.1.8
+  - https://aka.ms/automl-resources/packages/en_core_web_sm-2.1.0.tar.gz
```
Comment (lostmygithubaccount) [inline tutorials/quickstarts/environment.yml:27]: what is this? does not seem like a public pypi package

File: tutorials/quickstarts/score.py [hunk 14]
```diff
@@ -0,0 +1,22 @@
+import json
+import numpy as np
+import os
+import pickle
```
Comment (lostmygithubaccount) [inline tutorials/quickstarts/score.py:4]: unused imports

File: tutorials/quickstarts/sklearn-mnist-batch/train.py [hunk 15]
```diff
@@ -0,0 +1,82 @@
+import argparse
```
Comment (lostmygithubaccount) [inline tutorials/quickstarts/sklearn-mnist-batch/train.py:1]: consider refactoring into functions

File: tutorials/quickstarts/sklearn-mnist-batch/utils.py [hunk 16]
```diff
@@ -0,0 +1,27 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
```
Comment (lostmygithubaccount) [inline tutorials/quickstarts/sklearn-mnist-batch/utils.py:1]: remove license - this file can probably just be a cell in the relevant notebook(s)

File: tutorials/quickstarts/utils.py [hunk 17]
```diff
@@ -0,0 +1,27 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
```
Comment (lostmygithubaccount) [inline tutorials/quickstarts/utils.py:2]: duplicate file

File: tutorials/quickstarts/validated_linux_requirements.txt [hunk 18]
```diff
@@ -0,0 +1,119 @@
+adal<=1.2.6
```
Comment (lostmygithubaccount) [inline tutorials/quickstarts/validated_linux_requirements.txt:1]: this is a scary list of pinned dependencies

**Review by lostmygithubaccount — CHANGES_REQUESTED** [review]
main changes needed are:

- removing license cells 
- removing duplicate file(s) 

Some discussion needed probably with Data Science team for further sign off.

File: tutorials/quickstarts/automl_setup_linux.sh (hunk 10 above)
Comment (cecilidw) [reply tutorials/quickstarts/automl_setup_linux.sh]: Deleted this file, not needed.

File: tutorials/quickstarts/check_conda_version.py (hunk 11 above)
Comment (cecilidw) [reply tutorials/quickstarts/check_conda_version.py]: Deleted this file, not needed.

File: tutorials/quickstarts/utils.py (hunk 17 above)
Comment (cecilidw) [reply tutorials/quickstarts/utils.py:2]: Deleted this file, not needed.

File: tutorials/quickstarts/validated_linux_requirements.txt (hunk 18 above)
Comment (cecilidw) [reply tutorials/quickstarts/validated_linux_requirements.txt]: Deleted this file, not needed.

File: .github/workflows/tutorial-quickstarts.yml [hunk 19]
```diff
@@ -0,0 +1,56 @@
...
+jobs:
+  build:
+    runs-on: ubuntu-latest 
+    steps:
+    - name: check out repo
+      uses: actions/checkout@v2
+    - name: setup python
+      uses: actions/setup-python@v2
+      with:
+        python-version: "3.6"
```
[the remark this answers is not in the document]
Comment (cecilidw) [reply .github/workflows/tutorial-quickstarts.yml:19]: It was the version defined in the Compute Instance "Python 3.6 AzureML" environment, which was why I added it. I'm not sure if it is required, and it could probably be removed.

File: tutorials/quickstarts/sklearn-mnist-batch/utils.py (hunk 16 above)
Comment (cecilidw) [reply tutorials/quickstarts/sklearn-mnist-batch/utils.py]: License removed.

File: .github/workflows/tutorial-quickstarts.yml [hunk 20]
```diff
@@ -0,0 +1,56 @@
...
+    - uses: conda-incubator/setup-miniconda@v2
+      with:
+        activate-environment: azureml_py36
+        environment-file: tutorials/quickstarts/environment.yml
+        python-version: 3.6
+        auto-activate-base: false
+    - name: Conda info
+      shell: bash -l {0}
+      run: conda info
+    - name: Pip install ++
```
[the remark this answers is not in the document]
Comment (cecilidw) [reply .github/workflows/tutorial-quickstarts.yml:39]: If you mean the pip installs, they are included in the conda environment since it is the one activated in the shell at the time. But I could add them as part of the environment.yml, the reason I didn't was that I thought it might be an improvement to download the environment file from the Github location where I found it, to make sure it is always up to date.

File: .github/workflows/tutorial-quickstarts.yml [hunk 21]
```diff
@@ -0,0 +1,56 @@
...
+      shell: bash -l {0}
+      run: |
+        pip install azureml-opendatasets requests uuid
+        python -m ipykernel install --user --name azureml_py36 --display-name "azureml 36"
+        jupyter nbextension uninstall --user --py azureml.widgets
+    - name: run ClassificationWithAutomatedML.ipynb
```
[the remark this answers is not in the document]
Comment (cecilidw) [reply .github/workflows/tutorial-quickstarts.yml:45]: Yes, I agree. In hindsight I wondered if these should not be one tutorial but single notebooks with separate workflow files too.

File: README.md [hunk 22]
```diff
@@ -61,6 +61,7 @@ path|status|notebooks|description
...
 [deploy-triton](tutorials/deploy-triton)|[![deploy-triton](https://github.com/Azure/azureml-examples/workflows/tutorial-deploy-triton/badge.svg)](https://github.com/Azure/azureml-examples/actions?query=workflow%3Atutorial-deploy-triton)|[1.densenet-local.ipynb](tutorials/deploy-triton/1.densenet-local.ipynb)<br>[2.bidaf-aks-v100.ipynb](tutorials/deploy-triton/2.bidaf-aks-v100.ipynb)|learn how to efficiently deploy to GPUs using [triton inference server](https://github.com/triton-inference-server/server)
+[quickstarts](tutorials/quickstarts)|[![quickstarts](https://github.com/Azure/azureml-examples/workflows/tutorial-quickstarts/badge.svg)](https://github.com/Azure/azureml-examples/actions?query=workflow%3Atutorial-quickstarts)|[AzureMLin10mins.ipynb](tutorials/quickstarts/AzureMLin10mins.ipynb)<br>[ClassificationWithAutomatedML.ipynb](tutorials/quickstarts/ClassificationWithAutomatedML.ipynb)<br>[GettingStartedWithPythonSDK.ipynb](tutorials/quickstarts/GettingStartedWithPythonSDK.ipynb)|Quickstarts for use in Compute Instance
```
[the remark this answers is not in the document]
Comment (cecilidw) [reply README.md:64]: Changed the text to reflect comment :)

File: tutorials/quickstarts/AzureMLin10mins.ipynb [hunk 23]
```diff
@@ -0,0 +1,642 @@
+{
+  "cells": [
+    {
+      "cell_type": "markdown",
+      "source": [
+        "Copyright (c) Microsoft Corporation. All rights reserved. \n",
```
[the remark this answers is not in the document]
Comment (cecilidw) [reply tutorials/quickstarts/AzureMLin10mins.ipynb]: Removed.

File: tutorials/quickstarts/AzureMLin10mins.ipynb [hunk 24]
```diff
@@ -0,0 +1,642 @@
...
+        "\n",
+        "Licensed under the MIT License."
+      ],
+      "metadata": {
```
[the remark this answers is not in the document]
Comment (cecilidw) [reply tutorials/quickstarts/AzureMLin10mins.ipynb]: I copied the MD cells with license info from other notebooks (thought it was supposed to be there), so I'm not sure why it is there.

File: tutorials/quickstarts/AzureMLin10mins.ipynb [hunk 25]
```diff
@@ -0,0 +1,642 @@
...
+        "## Import Data\n",
+        "\n",
+        "Before you train a model, you need to understand the data that you are using to train it. In this section you learn how to:\n",
+        "\n",
+        "* Download the MNIST dataset\n",
+        "* Display some sample images\n",
+        "\n",
+        "### Download the MNIST dataset\n",
+        "\n",
+        "Use Azure Open Datasets to get the raw MNIST data files. [Azure Open Datasets](https://docs.microsoft.com/azure/open-datasets/overview-what-are-open-datasets) are curated public datasets that you can use to add scenario-specific features to machine learning solutions for more accurate models. Each dataset has a corrseponding class, `MNIST` in this case, to retrieve the data in different ways.\n",
```
[the remark this answers is not in the document]
Comment (cecilidw) [reply tutorials/quickstarts/AzureMLin10mins.ipynb:86]: Hm, yes, I wasn't aware that they aren't well supported. I haven't had any issues, so I think we should be good here.

File: tutorials/quickstarts/AzureMLin10mins.ipynb [hunk 26]
```diff
@@ -0,0 +1,642 @@
...
+        }
+      }
+    },
+    {
+      "cell_type": "markdown",
+      "source": [
+        "### Take a look at the data\n",
+        "Load the compressed files into `numpy` arrays. Then use `matplotlib` to plot 30 random images from the dataset with their labels above them. \n",
+        "\n",
+        "Note this step requires a `load_data` function that's included in an `utils.py` file. This file is placed in the same folder as this notebook. The `load_data` function simply parses the compressed files into numpy arrays."
```
[the remark this answers is not in the document]
Comment (cecilidw) [reply tutorials/quickstarts/AzureMLin10mins.ipynb:142]: I agree, and I will move it into the notebook.

File: tutorials/quickstarts/AzureMLin10mins.ipynb [hunk 27]
```diff
@@ -0,0 +1,642 @@
...
+          "transient": {
+            "deleting": false
+          }
+        }
+      }
+    },
+    {
+      "cell_type": "markdown",
+      "source": [
+        "## Version control your models with the Model Registry\n",
```
[the remark this answers is not in the document]
Comment (cecilidw) [reply tutorials/quickstarts/AzureMLin10mins.ipynb:286]: I appreciate nit and will fix it :)

File: tutorials/quickstarts/ClassificationWithAutomatedML.ipynb (hunk 2 above)
Comment (cecilidw) [reply tutorials/quickstarts/ClassificationWithAutomatedML.ipynb]: Removed.

File: tutorials/quickstarts/ClassificationWithAutomatedML.ipynb (hunk 3 above)
Comment (cecilidw) [reply tutorials/quickstarts/ClassificationWithAutomatedML.ipynb:121]: I agree.

File: tutorials/quickstarts/GettingStartedWithPythonSDK.ipynb (hunk 4 above)
Comment (cecilidw) [reply tutorials/quickstarts/GettingStartedWithPythonSDK.ipynb]: Removed.

File: tutorials/quickstarts/GettingStartedWithPythonSDK.ipynb (hunk 5 above)
Comment (cecilidw) [reply tutorials/quickstarts/GettingStartedWithPythonSDK.ipynb:324]: Removed "/en-us/" from all links, thanks for pointing out! Should there be a general rule or something similar that says try to avoid having links in the notebooks that are likely to break?

File: tutorials/quickstarts/GettingStartedWithPythonSDK.ipynb (hunk 6 above)
Comment (cecilidw) [reply tutorials/quickstarts/GettingStartedWithPythonSDK.ipynb:364]: Changed the links.

File: tutorials/quickstarts/GettingStartedWithPythonSDK.ipynb (hunk 7 above)
Comment (cecilidw) [reply tutorials/quickstarts/GettingStartedWithPythonSDK.ipynb:490]: Changed the links.

File: tutorials/quickstarts/README.md (hunk 8 above)
Comment (cecilidw) [reply tutorials/quickstarts/README.md:10]: Will do.

File: tutorials/quickstarts/README.md (hunk 9 above)
Comment (cecilidw) [reply tutorials/quickstarts/README.md:14]: Will do.

File: tutorials/quickstarts/environment.yml (hunk 12 above)
Comment (cecilidw) [reply tutorials/quickstarts/environment.yml:4]: Not sure, I got the file from https://github.com/Azure/MachineLearningNotebooks/tree/master/how-to-use-azureml/automated-machine-learning , as I though that must be the place to go for the correct version. I can check with the people that made it.

File: tutorials/quickstarts/environment.yml (hunk 13 above)
Comment (cecilidw) [reply tutorials/quickstarts/environment.yml:27]: Same as above, I can check with the owners of the how-to.

File: tutorials/quickstarts/score.py (hunk 14 above)
Comment (cecilidw) [reply tutorials/quickstarts/score.py:4]: Will remove it.

**Review by lostmygithubaccount — APPROVED** [review]

**Review by samuel100 — APPROVED** [review]
